### PR TITLE
Updated concept of "Configuration Profile" and Inspector Cleanup

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Change Request

## Overview

There was some confusion about the difference between a configurations and profiles, so this rename should help clear that up.

I also updated all the profile inspectors to property utilize `nameof(serializedProperty)` and use a common header path so updating it to use breadcrumb later can be easier.

## Breaking Changes

- `MixedRealityConfigurationProfile` -> `MixedRealityTookitRootProfile`
- `MixedRealityToolkit.ResetConfiguration(MixedRealityConfigurationProfile profile)` -> `MixedRealityToolkit.ResetProfile(MixedRealityTookitRootProfile profile)`
- `IMixedRealityServiceConfiguration.ConfigurationProfile` -> `IMixedRealityServiceConfiguration.Profile`

### CODE REVIEW

File: XRTK-Core/Packages/com.xrtk.core/Services/MixedRealityToolkit.cs [hunk 1]
```diff
@@ -59,12 +59,12 @@ public static bool HasActiveProfile
         /// </summary>
         [SerializeField]
         [Tooltip("The current active configuration for the Mixed Reality project")]
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Services/MixedRealityToolkit.cs:61]:
```suggestion
        [Tooltip("The current active settings for the Mixed Reality project")]
```
```suggestion
        [Tooltip("The current active configuration for the Mixed Reality project")]
```

**Comment by Jarodshow** [comment]
What about `MixedRealityRootProfile`? Or `MixedRealitySystemProfile`?

File: XRTK-Core/Packages/com.xrtk.core/Definitions/MixedRealityToolkitRootProfile.cs [hunk 2]
```diff
@@ -20,12 +20,12 @@
 namespace XRTK.Definitions
 {
     /// <summary>
-    /// Configuration profile settings for the Mixed Reality Toolkit.
+    /// The root profile for the Mixed Reality Toolkit's settings.
     /// </summary>
-    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Mixed Reality Toolkit Configuration Profile", fileName = "MixedRealityToolkitConfigurationProfile", order = (int)CreateProfileMenuItemIndices.Configuration)]
-    public class MixedRealityToolkitConfigurationProfile : BaseMixedRealityProfile
+    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Mixed Reality Toolkit Root Profile", fileName = "MixedRealityToolkitRootProfile", order = (int)CreateProfileMenuItemIndices.Configuration)]
+    public class MixedRealityToolkitRootProfile : BaseMixedRealityProfile
```
Comment (Jarodshow) [inline XRTK-Core/Packages/com.xrtk.core/Definitions/MixedRealityToolkitRootProfile.cs:26]: Should this be sealed?

**Comment by StephenHodgson** [comment]
Okay this Pull Request is ready

File: XRTK-Core/Packages/com.xrtk.core/Definitions/MixedRealityToolkitRootProfile.cs [hunk 3]
```diff
@@ -20,12 +20,12 @@
 namespace XRTK.Definitions
 {
     /// <summary>
-    /// Configuration profile settings for the Mixed Reality Toolkit.
+    /// The root profile for the Mixed Reality Toolkit's settings.
     /// </summary>
-    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Mixed Reality Toolkit Configuration Profile", fileName = "MixedRealityToolkitConfigurationProfile", order = (int)CreateProfileMenuItemIndices.Configuration)]
-    public class MixedRealityToolkitConfigurationProfile : BaseMixedRealityProfile
+    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Mixed Reality Toolkit Root Profile", fileName = "MixedRealityToolkitRootProfile", order = (int)CreateProfileMenuItemIndices.Configuration)]
+    public sealed class MixedRealityToolkitRootProfile : BaseMixedRealityProfile
```
Comment (Jarodshow) [inline XRTK-Core/Packages/com.xrtk.core/Definitions/MixedRealityToolkitRootProfile.cs:26]: Do we need the toolkit prefix?

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Definitions/MixedRealityToolkitRootProfile.cs:26]: Well it's the root profile for the toolkit, so I think it makes sense in this case.

**Comment by StephenHodgson** [comment]
Sorry guys one last review. I had to merge in the upstream changes

**Pull request merged**